src/core/inductive.ml

```diff
@@ -210,22 +292,35 @@ let gen_rec_type : Sig_state.t -> popt -> sym -> sym list -> term =
     inductive type T (i.e. xj = T v1 ... vm).
     Note: There cannot be name clashes between pattern variable names and
     function symbols names since pattern variables are prefixed by $. *)
-let gen_rec_rules : popt -> sym -> sym -> sym list -> p_rule list =
-  fun pos ind_sym rec_sym cons_list ->
+let gen_rec_rules :
+      popt -> (sym * (sym list) * sym) list ->
+      (sym * (tvar * tbox)) list -> p_rule list list =
+  fun pos ind_list assoc_predicates ->
```

Comment:
Would it be possible to simply define:
```
let gen_rec_rules : popt -> sym_pred_map -> sym * sym list * sym -> p_rule list
```
and later call `List.map (gen_rec_rules popt sym_pred_map) ind_rec_list`.

Reply:
I don't understand where you see `sym_pred_map`.

Reply:
This is a type I proposed to introduce in another comment. Search on https://github.com/Deducteam/lambdapi/pull/445.

Reply:
Yes this one: `(sym * (tvar * (tbox * tbox))) list`

Reply:
So `sym_pred_map = (sym * (tvar * (tbox * tbox))) list` or ` (sym * (tvar * tbox)) list`?

Reply:
`sym_pred_map = (sym * (tvar * (tbox * tbox))) list`